pkg/operations/cordondrainvm.go

```diff
@@ -173,24 +174,31 @@ func (o *drainOperation) deleteOrEvictPods(pods []v1.Pod) error {
 }
 
 func (o *drainOperation) evictPods(pods []v1.Pod, policyGroupVersion string) error {
+	ctx, cancel := context.WithCancel(context.Background())
 	doneCh := make(chan bool, len(pods))
 	errCh := make(chan error, 1)
 
 	for _, pod := range pods {
-		go func(pod v1.Pod, doneCh chan bool, errCh chan error) {
+		go func(ctx context.Context, pod v1.Pod, doneCh chan bool, errCh chan error) {
 			var err error
+		doneEviction:
```

Comment:
Why are we creating this additional named scope wrapping around the for loop?

Reply:
A "break" statement terminates execution of the innermost "for", "switch" or "select" statement. So in this case, we not only want to break the "select" statement, but also break the outer "for" statement. For more information: https://stackoverflow.com/questions/11104085/does-a-break-statement-break-from-a-switch-select

Reply:
> The scope of a label is the body of the function in which it is declared and excludes the body of any nested function.

Ref: https://golang.org/ref/spec#Label_scopes

`break doneEviction` in this context will break out of the entire go routine. See this go playground example:

https://play.golang.org/p/8MsyEUW1c4m

Reply:
The article you mentioned means that a label is not effective outside of the function where the label is defined if I understand it correctly.
You will have to use `WaitGroup` to see the results from goroutines, otherwise, the main method will quit before you see any results from goroutines.
Another thing is, in this PR, the label is in an inner `witch` statement.
Please see this go playground example:
https://play.golang.org/p/q6bBupZk5qg

Reply:
You mean `select` instead of "switch", right?

Reply:
yes, sorry, `select` is what I meant.

Reply:
And if you don't use the label, like this: https://play.golang.org/p/XanDbk87yNX. you will get an infinite for loop.

Reply:
All this was due to my misunderstanding of how for labels work (sorry about that), in addition to overly hasty go playground usage.

So I think we're good here.